Main.hs

```diff
@@ -54,7 +54,7 @@ faces (c:curves) = concat [ printf "f %d %d %d\nf %d %d %d\n" a b c b d c  | (a,
           height = 1 + length curves 
           width = length c
 
-(^*) f n = selfComposeN n f 
+(^*) f n = selfComposeN n f
```

Comment:
Great contribution ❤️

Reply:
lmao turns out those 'unnecessary spaces' I removed in the earlier PR weren't so `unnecessary`